### PR TITLE
Feature/nats and kafka libs

### PR DESCRIPTION
**Pull request by ajax-makarytskyi-o**
Added NATS and Kafka libs to project.

### CODE REVIEW

File: domainservice/src/main/kotlin/com/makarytskyi/rentcar/config/NatsConfiguration.kt
Comment (ajax-torbin-v) [inline domainservice/src/main/kotlin/com/makarytskyi/rentcar/config/NatsConfiguration.kt:1]: Do we need it, when using nats-lib?

File: domainservice/src/main/kotlin/com/makarytskyi/rentcar/kafka/CreateOrderKafkaProcessor.kt [hunk 1]
```diff
@@ -1,47 +1,36 @@
...
+import systems.ajax.kafka.handler.subscription.topic.TopicSingle
 
 @Component
-class CreateOrderKafkaProcessor(
-    private val createOrderKafkaReceiver: KafkaReceiver<String, ByteArray>,
-    private val natsConnection: Connection,
-) {
+class CreateOrderKafkaProcessor : KafkaHandler<Order, TopicSingle> {
+    @Autowired
+    private lateinit var natsConnection: Connection
```
Comment (ajax-torbin-v) [inline domainservice/src/main/kotlin/com/makarytskyi/rentcar/kafka/CreateOrderKafkaProcessor.kt]: Use NatsMessagePublisher

File: domainservice/src/main/kotlin/com/makarytskyi/rentcar/controller/nats/order/FindAllOrdersNatsController.kt [hunk 2]
```diff
@@ -4,29 +4,37 @@ import com.google.protobuf.Parser
...
+        orderService.findAll(inMsg.page, inMsg.size)
             .collectList()
             .map { it.toFindAllResponse() }
             .onErrorResume { it.toFindAllFailureResponse().toMono() }
 
+    override fun doOnUnexpectedError(inMsg: FindAllOrdersRequest?, e: Exception): Mono<FindAllOrdersResponse> =
+        FindAllOrdersResponse.newBuilder()
+            .apply {
+                failureBuilder.message = e.message
+            }.build().toMono()
```
Comment (ajax-marchenko-s) [inline domainservice/src/main/kotlin/com/makarytskyi/rentcar/controller/nats/order/FindAllOrdersNatsController.kt]: you could use here mapper from `ExceptionProtoMapper.kt`

File: domainservice/src/main/kotlin/com/makarytskyi/rentcar/kafka/CreateRepairingKafkaProducer.kt [hunk 3]
```diff
@@ -2,27 +2,22 @@ package com.makarytskyi.rentcar.kafka
...
-                    repairing.carId,
-                    repairing.toByteArray()
-                ),
-                null
-            ).toMono()
+        publisher.publish(
+            KafkaTopic.Repairing.REPAIRING_CREATE,
+            repairing.carId,
+            repairing,
+            KafkaPublisherOptions()
```
Comment (ajax-torbin-v) [inline domainservice/src/main/kotlin/com/makarytskyi/rentcar/kafka/CreateRepairingKafkaProducer.kt]: Do we need this options?

File: domainservice/src/test/kotlin/com/makarytskyi/rentcar/controller/CreateOrderNatsControllerTest.kt [hunk 4]
```diff
@@ -36,7 +41,7 @@ class CreateOrderNatsControllerTest : AbstractOrderNatsControllerTest() {
         ).toBuilder().successBuilder.orderBuilder.clearId().build()
 
         // WHEN
-        val response = sendRequest(CREATE, protoRequest, CreateOrderResponse.parser())
+        val response = natsPublisher.request(CREATE, protoRequest, CreateOrderResponse.parser()).block()!!
 
         // THEN
         assertNotNull(response.success.order.id)
```
Comment (ajax-torbin-v) [inline domainservice/src/test/kotlin/com/makarytskyi/rentcar/controller/CreateOrderNatsControllerTest.kt]: Provide message for AssertNotNull

**Review by ajax-marchenko-s — COMMENTED** [review]
Great work!

File: domainservice/src/main/kotlin/com/makarytskyi/rentcar/config/NatsConfiguration.kt
Comment (ajax-makarytskyi-o) [reply domainservice/src/main/kotlin/com/makarytskyi/rentcar/config/NatsConfiguration.kt:1]: No, I checked connection status. I fixed it.

File: domainservice/src/main/kotlin/com/makarytskyi/rentcar/kafka/CreateOrderKafkaProcessor.kt [hunk 5]
```diff
@@ -1,68 +1,58 @@
...
+                else
+                    IllegalStateException("NATS connection is closed").toMono()
             }
     }
 
-    private fun retryOnNatsConnection(): Retry {
-        return Retry.fixedDelay(Long.MAX_VALUE, Duration.ofSeconds(2))
-            .filter { throwable -> throwable is IllegalStateException }
-    }
+    private fun retryOnNatsError(): Retry = Retry.fixedDelay(Long.MAX_VALUE, Duration.ofSeconds(2))
```
Comment (ajax-shulepa-a) [inline domainservice/src/main/kotlin/com/makarytskyi/rentcar/kafka/CreateOrderKafkaProcessor.kt]: should it be fixed delay?
btw in kafka-lib there's already retry mechanism

**Review by ajax-shulepa-a — APPROVED** [review]

File: domainservice/src/main/kotlin/com/makarytskyi/rentcar/config/NatsListener.kt [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package com.makarytskyi.rentcar.config
+
+import io.nats.client.Connection
+import io.nats.client.ConnectionListener
+import org.springframework.stereotype.Component
+
+@Component
+class NatsListener : ConnectionListener {
```
Comment (ajax-shulepa-a) [inline domainservice/src/main/kotlin/com/makarytskyi/rentcar/config/NatsListener.kt:8]: why do you need this class?

Comment (ajax-makarytskyi-o) [reply domainservice/src/main/kotlin/com/makarytskyi/rentcar/config/NatsListener.kt:8]: To check status of connection to NATS, if it is disconnected, I stop consume from kafka topic items and wait until it is connected.

**Pull request merged**